httpdocs/js/webots-cloud.js

```diff
@@ -1250,6 +1486,304 @@ ${deleteProject}`;
+      const fieldRegex = /\[\n((.*\n)*)\]/mg;
+      let matches = proto.matchAll(fieldRegex);
+      let fieldsDefinition;
+      const fieldEnumeration = new Map();
+      const describedField = [];
+      let fields = '';
+      let file = '';
+      for (const match of matches) {
+        fieldsDefinition = match[1];
+        break;
+      }
+
+      // remove enumerations
+      const removeEnumRegex = /.*ield\s+([^ ]*?)(\{(?:[^\[\n]*\,?\s?)(?<!(\{))\})\s+([^ ]*)\s+([^#\n]*)(#?)(.*)/mg;
+      matches = fieldsDefinition.matchAll(removeEnumRegex);
+      for (const match of matches) {
+        fieldEnumeration.set(match[4], match[2].slice(1, -1).split(','));
+        if (match[0].includes('\n')) {
+          const string = ' '.repeat(match[0].indexOf(match[2]));
+          fieldsDefinition = fieldsDefinition.replace(string + match[4], match[4]);
+          fieldsDefinition = fieldsDefinition.replace(match[2] + '\n', '');
+        }
+
+        if (match[2].length < 40)
+          fieldsDefinition = fieldsDefinition.replace(match[2], ' '.repeat(match[2].length));
+        else
+          fieldsDefinition = fieldsDefinition.replace(match[2], '');
+      }
+
+      const spacingRegex = /.*ield\s+([^ ]*?)(\s+)([^ ]*)\s+([^#\n]*)(#?)(.*)/mg;
+      matches = fieldsDefinition.matchAll(spacingRegex);
+      let minSpaces = 2000;
+      for (const match of matches) {
+        let spaces = match[2];
+        if (spaces.length < minSpaces)
+          minSpaces = spaces.length;
+      }
+      const spacesToRemove = Math.max(minSpaces - 2, 0);
+
+      const cleaningRegex = /^\s*(.*?ield)\s+([^ \{]*)(\s+)([^ ]*)\s+([^#\n]*)(#?)(.*)((\n*( {4}| {2}\]).*)*)/gm;
+      const isDescriptionRegex = /Is\s`([a-zA-Z]*).([a-zA-Z]*)`./g;
+
+      const baseNodeList = ['WorldInfo', 'Hinge2JointParameters', 'PBRAppearance', 'ContactProperties', 'SolidReference',
+        'Charger', 'Capsule', 'Mesh', 'Background', 'BallJoint', 'Focus', 'RotationalMotor', 'ElevationGrid', 'Pen',
+        'Cylinder', 'GPS', 'SliderJoint', 'Compass', 'Emitter', 'Track', 'Cone', 'LED', 'Slot', 'Radar', 'Coordinate',
+        'HingeJointParameters', 'Hinge2Joint', 'LinearMotor', 'Sphere', 'JointParameters', 'TrackWheel', 'Appearance',
+        'HingeJoint', 'DirectionalLight', 'Accelerometer', 'Viewpoint', 'Speaker', 'IndexedLineSet', 'PointSet', 'Damping',
+        'ImmersionProperties', 'Robot', 'Lidar', 'DistanceSensor', 'Camera', 'Lens', 'Altimeter', 'Color', 'Transform',
+        'Recognition', 'Connector', 'Propeller', 'LensFlare', 'BallJointParameters', 'TextureTransform', 'IndexedFaceSet',
+        'Normal', 'Fog', 'Display', 'TouchSensor', 'Shape', 'TextureCoordinate', 'Box', 'ImageTexture', 'Radio', 'CadShape',
+        'Plane', 'RangeFinder', 'Physics', 'SpotLight', 'Brake', 'PointLight', 'PositionSensor', 'Zoom', 'InertialUnit',
+        'LightSensor', 'Gyro', 'Receiver', 'Microphone', 'Solid', 'Billboard', 'Fluid', 'Muscle', 'Group', 'Skin',
+        'Material'];
```

Comment:
That will be difficult to maintain on the long run...
Did you updated the wiki explaining how to add a new node so that we also maintain this list?